### PR TITLE
Add new args for CORS

### PR DESCRIPTION
**Pull request by yuzhenmi**

### CODE REVIEW

File: start_esp/nginx-auto.conf.template [hunk 1]
```diff
@@ -113,6 +113,24 @@ http {
 % endif
 
 % if location.proto == 'grpc':
+  % if cors_preset == 'basic':
```
Comment (qiwzhang) [inline start_esp/nginx-auto.conf.template]: I will like to make this cors support not grpc specific?  it can also support http too.

**Comment by qiwzhang** [comment]
@lizan, @lei-tang could you help to review this?

**Comment by lei-tang** [comment]
@qiwzhang Yes, will review it now.

File: start_esp/nginx-auto.conf.template [hunk 2]
```diff
@@ -113,6 +113,24 @@ http {
...
+      # Begin basic CORS preset for GRPC backend
+      if (${'$'}request_method = 'OPTIONS') {
+        add_header 'Access-Control-Max-Age' 1728000;
```
Comment (lei-tang) [inline start_esp/nginx-auto.conf.template]: Consider making 'Access-Control-Max-Age' a customizable parameter, similar to other input parameters such as cors_allow_origin, cors_allow_methods, and cors_allow_headers.

File: start_esp/start_esp.py [hunk 3]
```diff
@@ -600,6 +605,53 @@ def make_argparser():
...
+        backend cannot set CORS headers.
+        Specify one of available presets to configure CORS response headers
+        in nginx. Defaults to no preset and therefore no CORS response
+        headers. If no preset is suitable for the use case, use the
+        --nginx_config arg to use a custom nginx config file.
+        Available presets:
+        - basic - Assumes all location paths have the same CORS policy.
+          Responds to preflight OPTIONS requests with an empty 204, and the
+          results of preflight are allowed to be cached for up to 20 days
+          (1728000 seconds). See descriptions for args --cors_allow_origin,
```
Comment (lei-tang) [inline start_esp/start_esp.py:629]: Consider making the preflight caching time a customizable parameter, with a default value.

File: start_esp/nginx-auto.conf.template (hunk 1 above)
Comment (yuzhenmi) [reply start_esp/nginx-auto.conf.template]: Yes, I'll move this block out of grpc when I get the time.

**Review by qiwzhang — APPROVED** [review]

**Pull request merged**